android/app/src/main/java/com/mattermost/helpers/Credentials.java

```diff
@@ -1,26 +1,70 @@
 package com.mattermost.helpers;
 
 import androidx.annotation.Nullable;
+import androidx.biometric.BiometricPrompt;
+import androidx.fragment.app.FragmentActivity;
 
 import com.facebook.react.bridge.Arguments;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.WritableMap;
 import com.oblador.keychain.KeychainModule;
 
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 
 public class Credentials {
-    public static void getCredentialsForServer(ReactApplicationContext context, String serverUrl, ResolvePromise promise) {
+
+    public static void getCredentialsForServer(ReactApplicationContext context, String serverUrl, ResolvePromise promise, boolean useBiometrics) {
         final KeychainModule keychainModule = new KeychainModule(context);
 
-        final WritableMap options = Arguments.createMap();
-        // KeyChain module fails if `authenticationPrompt` is not set
-        final WritableMap authPrompt = Arguments.createMap();
-        authPrompt.putString("title", "Authenticate to retrieve secret");
-        authPrompt.putString("cancel", "Cancel");
-        options.putMap("authenticationPrompt", authPrompt);
-        options.putString("service", serverUrl);
+        if (useBiometrics) {
+            authenticateUserWithBiometrics(context, new BiometricPrompt.AuthenticationCallback() {
+                @Override
+                public void onAuthenticationSucceeded(BiometricPrompt.AuthenticationResult result) {
+                    WritableMap options = Arguments.createMap();
+                    WritableMap authPrompt = Arguments.createMap();
+                    authPrompt.putString("title", "Authenticate to retrieve secret");
+                    authPrompt.putString("cancel", "Cancel");
+                    options.putMap("authenticationPrompt", authPrompt);
+                    options.putString("service", serverUrl);
+
+                    keychainModule.getGenericPasswordForOptions(options, promise);
```

Comment:
This block is the same as the block in lines 45-52. Any chance we can refactor it out?